src/solver/simulation/timeseries-numbers.cpp

```diff
@@ -187,6 +190,23 @@ static unsigned int CheckMatricesWidth(const Data::Study& study)
             }
             break;
         }
+        case Data::timeSeriesRenewable:
```

Comment:
I've run a few mental experiments on some width lists, here are the results.

Assume `TimeSeriesT = Data::timeSeriesLoad`. Very similar for clusters (thermal or renewable), the per-cluster lists are just concatenated into a big one, for which we can use the same reasoning as here.

### W=[1,1,1,1]
Four areas, each with a matrix of width 1. The function returns `w=1`.
### W=[2,2,2,2]
Four areas, each with a matrix of width 2. The function returns `w=2` after all iterations.
### W=[1,2,3,4]
Four areas, with matrices of width 1, 2, 3, 4 respectively. At the 3rd iteration, return 0.
### W=[4,3,2,1]
Four areas, with matrices of width 4, 3, 2, 1 respectively. At the 2nd iteration, return 0.

In conclusion, the function returns 0 whenever two width are different and different than 1. Otherwise, return the common value.

0 is an error value :
```cpp
nbTimeseries = CheckMatricesWidth<TimeSeriesT>(study);
if (!nbTimeseries)
{
    logs.error() << "Intra-modal correlation: Constraint violation: The number of time-series "
                    "must be identical for all areas";
    return false;
}
```

Reply:
More investigations. I wrote a small Python function to reproduce the behavior
```python3
def f(L):
    w = 1
    for p in L:
        if p != w:
            if p != 1 and w != 1:
                return 0
            if p > 1:
                w = p
    return w
```
For the testing part :
```
f([1, 1, 1, 1]) = 1
f([2, 2, 2, 2]) = 2
f([1, 2, 3, 4]) = 0
f([4, 3, 2, 1]) = 0
f([0, 0]) = 1
f([0, 1, 0]) = 1
f([0, 1, 2]) = 2
f([0, 2, 1]) = 2
f([2, 1, 0]) = 0 # PROBLEM : a permutation should not affect the result
```
There is a problem (see comment) on permutations.

Also, the function returns 1 when all matrices are empty (width=0), which is expected.